test/controllers/SelectedTransactionsControllerSpec.scala

```diff
@@ -36,137 +36,227 @@ import scala.concurrent.Future
+        fakeRequest(GET, routes.SelectedTransactionsController.onPageLoad().url)
+
+      running(app) {
+        val result = route(app, request).value
+        status(result) mustBe SEE_OTHER
+      }
     }
   }
 
-  "when too many results returned for the search" in new Setup {
-    val day = 30
-    val month = 3
-    val year = 2023
+  "OnSubmit" must {
+
+    "redirect to selected-confirmation page when cash account statement is submitted successfully" in new Setup {
 
-    when(mockRequestedTransactionsCache.get(any))
-      .thenReturn(Future.successful(Some(CashTransactionDates(
-        LocalDate.of(year, month, day), LocalDate.of(year, month, day)))))
+      when(mockRequestedTransactionsCache.get(any))
+        .thenReturn(Future.successful(Some(CashTransactionDates(LocalDate.now(), LocalDate.now()))))
 
-    when(mockCustomsFinancialsApiConnector.getCashAccount(eqTo(eori))(any, any))
-      .thenReturn(Future.successful(Some(cashAccount)))
+      when(mockCustomsFinancialsApiConnector.getCashAccount(eqTo(eori))(any, any))
+        .thenReturn(Future.successful(Some(cashAccount)))
 
-    when(mockCustomsFinancialsApiConnector.retrieveHistoricCashTransactions(eqTo(cashAccountNumber), any, any)(any))
-      .thenReturn(Future.successful(Left(TooManyTransactionsRequested)))
+      when(mockCustomsFinancialsApiConnector.postCashAccountStatementRequest(any, any, any, any)(any))
+        .thenReturn(Future.successful(Right(accountResCommon01)))
 
-    val request: FakeRequest[AnyContentAsEmpty.type] =
-      fakeRequest(GET, routes.SelectedTransactionsController.onPageLoad().url)
+      val request: FakeRequest[AnyContentAsEmpty.type] =
+        fakeRequest(POST, routes.SelectedTransactionsController.onSubmit().url)
 
-    running(app) {
-      val result = route(app, request).value
-      status(result) mustBe SEE_OTHER
+      running(app) {
+        val result = route(app, request).value
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.ConfirmationPageController.onPageLoad().url)
+      }
     }
+
+    "redirect to requested too many transactions page " when {
+
+      "cash account statement request responds with EXCEEDED_MAXIMUM in statusText" in new Setup {
+
+        when(mockRequestedTransactionsCache.get(any))
+          .thenReturn(Future.successful(Some(CashTransactionDates(LocalDate.now(), LocalDate.now()))))
+
+        when(mockCustomsFinancialsApiConnector.getCashAccount(eqTo(eori))(any, any))
+          .thenReturn(Future.successful(Some(cashAccount)))
+
+        when(mockCustomsFinancialsApiConnector.postCashAccountStatementRequest(any, any, any, any)(any))
+          .thenReturn(Future.successful(Left(ExceededMaximum)))
+
+        val request: FakeRequest[AnyContentAsEmpty.type] =
+          fakeRequest(POST, routes.SelectedTransactionsController.onSubmit().url)
+
+        running(app) {
+          val result = route(app, request).value
+          status(result) mustBe SEE_OTHER
```

Comment:
Better to give a line break before this line to distinguish between result and assertions